crates/bls-gadgets/src/utils.rs

```diff
@@ -1,11 +1,11 @@
-use algebra::Field;
-use r1cs_core::{ConstraintSystem, SynthesisError};
-use r1cs_std::{alloc::AllocGadget, boolean::Boolean};
-
+use algebra::{ 
+    PrimeField
+};
+use r1cs_std::{boolean::Boolean, R1CSVar};
 /// Helper used to skip operations which should not be executed when running the
 /// trusted setup
-pub fn is_setup(message: &[Boolean]) -> bool {
-    message.iter().any(|m| m.get_value().is_none())
+pub fn is_setup<F: PrimeField>(message: &[Boolean<F>]) -> bool {
```

Comment:
Maybe we can get the mode from the `cs` itself?

Reply:
Yes. Changed to this